tests/headingcommand.js

```diff
@@ -171,13 +158,13 @@ describe( 'HeadingCommand', () => {
 				convertTo = option;
 			}
 
-			it( 'converts to default option when executed with already applied option', () => {
+			it( 'does nothing when executed with already applied option', () => {
 				const command = commands.heading1;
 
 				setData( document, '<heading1>foo[]bar</heading1>' );
 				command.execute();
 
-				expect( getData( document ) ).to.equal( '<paragraph>foo[]bar</paragraph>' );
+				expect( getData( document ) ).to.equal( '<heading1>foo[]bar</heading1>' );
```

Comment:
It would be clean but it is not needed. The delta will be created and it will do nothing.

The only case where it could matter, that comes to my mind, is OT. If, at the same time, one client change heading to heading and the other change heading to paragraph, heading-to-heading change might be stronger and overwrite heading-to-paragraph change.